File: the_vine_hub_app/lib/core/network/api/discipleship/models/discipleship_note_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'discipleship_note_entry_dto.dart';

part 'discipleship_note_dto.freezed.dart';
part 'discipleship_note_dto.g.dart';

@freezed
abstract class DiscipleshipNoteDto with _$DiscipleshipNoteDto {
  const factory DiscipleshipNoteDto({
    required String noteId,
    String? title,
    String? description,
    int? noteStatus,
    required DateTime createdAt,
    List<String>? categories,
    String? discipleId,
    String? leaderId,
    List<DiscipleshipNoteEntryDto>? entries,
  }) = _DiscipleshipNoteDto;

  factory DiscipleshipNoteDto.fromJson(Map<String, dynamic> json) =>
      _$DiscipleshipNoteDtoFromJson(json);
}

File: backend/TheVineHub.API/Features/Discipleship/DiscipleshipCommon/DiscipleshipNoteDto.cs
using System.ComponentModel.DataAnnotations.Schema;
using TheVineHub.API.Features.Discipleship;
namespace TheVineHub.API.Features.Discipleship
{
    public class DiscipleshipNoteDto
    {
        [Column("note_id")]
        public string NoteId { get; set; } = string.Empty;
        [Column("title")]
        public required string Title { get; set; }
        [Column("description")]
        public string Description { get; set; } = string.Empty;
        [Column("note_status")]
        public NoteStatus NoteStatus { get; set; }
        [Column("created_at")]
        public DateTime CreatedAt { get; set; }
        [Column("categories")]
        public List<string> Categories { get; set; } = [];

        [Column("disciple_id")]

        public required string DiscipleId { get; set; }
        [Column("leader_id")]
        public required string LeaderId { get; set; }

        [Column("entries")]

        public IList<DiscipleshipNoteEntryDto> Entries { get; set; } = [];
    }
}
